### PR TITLE
scenario 4.1.1

### PR DESCRIPTION
**Pull request by NikitaAvraimov-okta**

### CODE REVIEW

**Comment by NikitaAvraimov-okta** [comment]
Want to discuss A18N test wrapper implementation, currently have local version made with okHttp, will replace in a follow up PR for self-registration today if thats okay. @JayNewstrom

File: app/src/androidTest/java/com/okta/idx/android/SelfServiceRegistrationTestEndToEnd.kt [hunk 1]
```diff
@@ -49,6 +50,45 @@ import org.junit.runner.RunWith
...
+        waitForElement(CODE_EDIT_TEXT)
+        val code = A18NWrapper.getCodeFromEmail(profile)
+        onView(withId(R.id.code_edit_text)).perform(replaceText(code))
+        onView(withId(R.id.submit_button)).perform(click())
+
+        waitForElement(SELECT_BUTTON)
+        onView(withId(R.id.skip_button)).perform(click())
+
+        waitForElement(ID_TOKEN_TYPE_TEXT_VIEW)
+        A18NWrapper.deleteProfile(profile)
```
Comment (JayNewstrom) [inline app/src/androidTest/java/com/okta/idx/android/SelfServiceRegistrationTestEndToEnd.kt:87]: If this test fails, you'll leak this profile.

**Review by JayNewstrom — CHANGES_REQUESTED** [review]

File: app/src/androidTest/java/com/okta/idx/android/SelfServiceRegistrationTestEndToEnd.kt (hunk 1 above)
Comment (NikitaAvraimov-okta) [reply app/src/androidTest/java/com/okta/idx/android/SelfServiceRegistrationTestEndToEnd.kt]: Added @After section

File: app/src/androidTest/java/com/okta/idx/android/SelfServiceRegistrationTestEndToEnd.kt [hunk 2]
```diff
@@ -50,18 +51,26 @@ import org.junit.runner.RunWith
 class SelfServiceRegistrationTestEndToEnd {
     @get:Rule val activityRule = ActivityScenarioRule(MainActivity::class.java)
 
+    var profile: A18NProfile? = null
```
Comment (JayNewstrom) [inline app/src/androidTest/java/com/okta/idx/android/SelfServiceRegistrationTestEndToEnd.kt:54]: Can be `private`.

File: app/src/androidTest/java/com/okta/idx/android/SelfServiceRegistrationTestEndToEnd.kt [hunk 3]
```diff
@@ -50,18 +51,26 @@ import org.junit.runner.RunWith
...
+            A18NWrapper.deleteProfile(it)
+        }
+    }
+
     // Mary signs up for an account with Password, sets up required Email factor, then skips
     // optional SMS.
     @Test fun scenario_4_1_1() {
-        val profile = A18NWrapper.createProfile()
-
+        profile = A18NWrapper.createProfile()
```
Comment (JayNewstrom) [inline app/src/androidTest/java/com/okta/idx/android/SelfServiceRegistrationTestEndToEnd.kt:66]: Can we do this in a `@Before`? The other tests will need this too.

Comment (NikitaAvraimov-okta) [reply app/src/androidTest/java/com/okta/idx/android/SelfServiceRegistrationTestEndToEnd.kt:66]: 4.1.3 does not use a18n profiles

Comment (NikitaAvraimov-okta) [reply app/src/androidTest/java/com/okta/idx/android/SelfServiceRegistrationTestEndToEnd.kt:66]: Should we still put this into `@Before`

Comment (JayNewstrom) [reply app/src/androidTest/java/com/okta/idx/android/SelfServiceRegistrationTestEndToEnd.kt:66]: Leaving it as is works given that information.

File: app/src/androidTest/java/com/okta/idx/android/SelfServiceRegistrationTestEndToEnd.kt [hunk 4]
```diff
@@ -49,6 +51,53 @@ import org.junit.runner.RunWith
...
+        waitForElement(CODE_EDIT_TEXT)
+        val code = A18NWrapper.getCodeFromEmail(profile!!)
+        onView(withId(R.id.code_edit_text)).perform(replaceText(code))
+        onView(withId(R.id.submit_button)).perform(click())
+
+        waitForElement(SELECT_BUTTON)
+        onView(withId(R.id.skip_button)).perform(click())
+
+        waitForElement(ID_TOKEN_TYPE_TEXT_VIEW)
+        A18NWrapper.deleteProfile(profile!!)
```
Comment (JayNewstrom) [inline app/src/androidTest/java/com/okta/idx/android/SelfServiceRegistrationTestEndToEnd.kt:96]: You're doing a double delete now.

File: app/src/androidTest/java/com/okta/idx/android/SelfServiceRegistrationTestEndToEnd.kt [hunk 5]
```diff
@@ -49,6 +51,52 @@ import org.junit.runner.RunWith
...
+
+    // Mary signs up for an account with Password, sets up required Email factor, then skips
+    // optional SMS.
+    @Test fun scenario_4_1_1() {
+        profile = A18NWrapper.createProfile()
+        activityRule.scenario.moveToState(Lifecycle.State.RESUMED)
+        onView(withId(R.id.self_service_registration_button)).perform(click())
+        waitForElement(FIRST_NAME_EDIT_TEXT)
+
+        onView(withId(R.id.first_name_edit_text)).perform(replaceText("Mary"))
```
Comment (JayNewstrom) [inline app/src/androidTest/java/com/okta/idx/android/SelfServiceRegistrationTestEndToEnd.kt:71]: This is going to flood our test org with users. Please leave this open until I have my PR for user management so we can delete these users.

**Pull request closed without merging**